<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>How ecological interactions shape microbial mutation rates to antimicrobial resistance - 2&nbsp; Working together to control mutation: how collective peroxide detoxification determines microbial mutation rate plasticity</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./Discussion.html" rel="next">
<link href="./Introduction.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./Chap2PBio.html"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Working together to control mutation: how collective peroxide detoxification determines microbial mutation rate plasticity</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">How ecological interactions shape microbial mutation rates to antimicrobial resistance</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Abbreviations</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Abstract.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Abstract</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Declaration.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Declaration</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Copyright.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Copyright</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Acknowledgements.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Acknowledgements</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./LayAbstract.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Simplified Abstract</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Chap2PBio.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Working together to control mutation: how collective peroxide detoxification determines microbial mutation rate plasticity</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Discussion.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Discussion</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./summary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Summary</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">2.1</span> Introduction</a></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results"><span class="header-section-number">2.2</span> Results</a>
  <ul class="collapse">
  <li><a href="#initial-computational-model-of-nucleotide-metabolism-in-a-growing-microbial-population-fails-to-reproduce-mutation-rate-plasticity" id="toc-initial-computational-model-of-nucleotide-metabolism-in-a-growing-microbial-population-fails-to-reproduce-mutation-rate-plasticity" class="nav-link" data-scroll-target="#initial-computational-model-of-nucleotide-metabolism-in-a-growing-microbial-population-fails-to-reproduce-mutation-rate-plasticity"><span class="header-section-number">2.2.1</span> Initial computational model of nucleotide metabolism in a growing microbial population fails to reproduce mutation rate plasticity</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Working together to control mutation: how collective peroxide detoxification determines microbial mutation rate plasticity</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="introduction" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="introduction"><span class="header-section-number">2.1</span> Introduction</h2>
<p>Uncovering the mechanisms behind environmentally responsive mutagenesis informs our understanding of evolution, notably antimicrobial resistance, where mutation supply can be critical <span class="citation" data-cites="Gifford2023 Ragheb2019">(<a href="#ref-Gifford2023" role="doc-biblioref">Gifford et al. 2023</a>; <a href="#ref-Ragheb2019" role="doc-biblioref">Ragheb et al. 2019</a>)</span>. Microbial mutation rates are responsive to a wide variety of environmental factors including population density <span class="citation" data-cites="Krasovec2017">(<a href="#ref-Krasovec2017" role="doc-biblioref">Krašovec et al. 2017</a>)</span>, temperature <span class="citation" data-cites="Chu2018">(<a href="#ref-Chu2018" role="doc-biblioref">Chu et al. 2018</a>)</span>, growth rate <span class="citation" data-cites="Maharjan2018 Liu2019">(<a href="#ref-Maharjan2018" role="doc-biblioref">Ram P. Maharjan and Ferenci 2018</a>; <a href="#ref-Liu2019" role="doc-biblioref">Liu and Zhang 2019</a>)</span>, stress <span class="citation" data-cites="MacLean2013 Foster2007">(<a href="#ref-MacLean2013" role="doc-biblioref">MacLean, Torres-Barceló, and Moxon 2013</a>; <a href="#ref-Foster2007" role="doc-biblioref">Foster 2007</a>)</span>, growth phase <span class="citation" data-cites="Loewe2003">(<a href="#ref-Loewe2003" role="doc-biblioref">Loewe, Textor, and Scherer 2003</a>)</span> and nutritional state <span class="citation" data-cites="Maharjan2017">(<a href="#ref-Maharjan2017" role="doc-biblioref">Ram P. Maharjan and Ferenci 2017</a>)</span>. Such mutation rate plasticity inspires the idea of “anti-evolution drugs”, able to slow the evolution of antimicrobial resistance during the treatment of an infection <span class="citation" data-cites="Ragheb2019 Cirz2005 Domenech2020 Alam2016">(<a href="#ref-Ragheb2019" role="doc-biblioref">Ragheb et al. 2019</a>; <a href="#ref-Cirz2005" role="doc-biblioref">Cirz et al. 2005</a>; <a href="#ref-Domenech2020" role="doc-biblioref">Domenech et al. 2020</a>; <a href="#ref-Alam2016" role="doc-biblioref">Alam et al. 2016</a>)</span>. Even small reductions in the mutation rate (2-5-fold) can have dramatic effects on the capacity of bacterial populations to adapt to antibiotic treatment, particularly when evolution is limited by mutation supply, as is the case for small pathogen populations <span class="citation" data-cites="Ragheb2019">(<a href="#ref-Ragheb2019" role="doc-biblioref">Ragheb et al. 2019</a>)</span>.</p>
<p>Microbial mutation rates have an inverse association with population density across all domains of life, we have previously shown that 93% of otherwise unexplained variation in published mutation rate estimates is explained by the final population density <span class="citation" data-cites="Krasovec2017">(<a href="#ref-Krasovec2017" role="doc-biblioref">Krašovec et al. 2017</a>)</span>. This density-associated mutation rate plasticity (DAMP) is a distinct phenotype from stress-induced mutagenesis, which acts via independent genetic mechanisms <span class="citation" data-cites="Krasovec2018">(<a href="#ref-Krasovec2018" role="doc-biblioref">Krašovec et al. 2018</a>)</span>. Population density alters not only the rate but also the spectrum of mutations, with significantly higher rates of AT&gt;GC transitions seen in low density populations <span class="citation" data-cites="Gifford2023">(<a href="#ref-Gifford2023" role="doc-biblioref">Gifford et al. 2023</a>)</span>. Density effects are likely relevant to natural populations given that population sizes and densities vary greatly, for example, <em>Escherichia coli</em> populations in host faeces can range in density by 5 orders of magnitude <span class="citation" data-cites="Tenaillon2010">(<a href="#ref-Tenaillon2010" role="doc-biblioref">Tenaillon et al. 2010</a>)</span>, and infections can be established by populations as small as 6×10<sup>3</sup> cells <span class="citation" data-cites="Cornick2004">(<a href="#ref-Cornick2004" role="doc-biblioref">Cornick and Helgerson 2004</a>)</span>. We therefore aim to mechanistically describe the widespread phenotype of DAMP.</p>
<p>In order to test potential mechanisms generating DAMP, we developed and systematically assessed a computational model connecting metabolism and mutagenesis in a growing <em>E. coli</em> population. This model generates the hypothesis that the key determinants of DAMP are the production and degradation rates of reactive oxygen species (ROS). Though molecular oxygen is relatively stable it can be reduced to superoxide (<sup>•</sup>O<sub>2−</sub>), hydrogen peroxide (H<sub>2</sub>O<sub>2</sub>) and hydroxyl radicals (HO<sup>•</sup>). These “reactive oxygen species” are strong oxidants able to damage multiple biological molecules including nucleotides and DNA <span class="citation" data-cites="Imlay2003">(<a href="#ref-Imlay2003" role="doc-biblioref">Imlay 2003</a>)</span>. We tested the role of ROS in controlling DAMP by estimating mutation rate plasticity under different conditions of environmental oxygen and with genetic manipulations known to alter ROS dynamics. We find that the reduction in mutation rate at increased population density results from the population’s increased ability to degrade H<sub>2</sub>O<sub>2</sub>, resulting in reduced ROS-associated mutagenesis. We show that this density effect is also experienced by cells deficient in H<sub>2</sub>O<sub>2</sub> degradation when cocultured with wild-type cells able to detoxify the environment. Mutation rates therefore depend not only on the genotype of the individual but also on the community’s capacity to degrade H<sub>2</sub>O<sub>2</sub>.</p>
</section>
<section id="results" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="results"><span class="header-section-number">2.2</span> Results</h2>
<section id="initial-computational-model-of-nucleotide-metabolism-in-a-growing-microbial-population-fails-to-reproduce-mutation-rate-plasticity" class="level3" data-number="2.2.1">
<h3 data-number="2.2.1" class="anchored" data-anchor-id="initial-computational-model-of-nucleotide-metabolism-in-a-growing-microbial-population-fails-to-reproduce-mutation-rate-plasticity"><span class="header-section-number">2.2.1</span> Initial computational model of nucleotide metabolism in a growing microbial population fails to reproduce mutation rate plasticity</h3>
<div id="fig-one" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ggpubr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:plyr':

    mutate</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:cowplot':

    get_legend</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Linking to ImageMagick 6.9.12.93
Enabled features: cairo, fontconfig, freetype, heic, lcms, pango, raw, rsvg, webp
Disabled features: fftw, ghostscript, x11</code></pre>
</div>
<div class="cell-output-display">
<div id="fig-one" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="Chap2PBio_files/figure-html/fig-one-1.png" class="img-fluid figure-img" data-ref-parent="fig-one" width="672"></p>
<figcaption class="figure-caption">(a) A: Model structure connecting variables. Red variables indicate the pathway to mutagenesis; green numbered arrows indicate pathways targeted by model variants. This structure was represented in ODEs, parameterised from the literature (Methods), and simulated to give output shown in B-D. B: Kinetics of external glucose concentration (eGlc), C: population size (wtCell divided by G nucleotides in the E. coli genome) and D: mutation rate (mCell/wtCell). Note log scale on y-axis in panel D. Panels B-D are plotted for 5 initial glucose concentrations (range 55 – 1100 mg L-1 as shown in legend), initial glucose concentration indicated by line colour.</figcaption>
</figure>
</div>
</div>
<figcaption class="figure-caption">Figure&nbsp;2.1: Dynamical computational model of growth, metabolism and mutagenesis in E. coli.</figcaption>
</figure>
</div>
<div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 10 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (5): Variable, Definition, Starting Value, Units, Region

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div id="tbl-one" class="cell tbl-parent quarto-layout-panel anchored">
<div class="panel-caption table-caption">
<p>Table&nbsp;2.1: Definitions and starting values for the 10 variables in ODE model A.</p>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div id="tbl-one-1" class="quarto-layout-cell quarto-layout-cell-subref anchored" data-ref-parent="tbl-one" style="flex-basis: 100.0%;justify-content: center;">
<table class="table table-sm table-striped small">
<caption>(a) For variables measured as a concentration, the volume within which this is calculated is given in the ‘region’ column. wtCell and cytVol starting values equate to 2175 cells (assuming 2357528 GC bp in the E. coli genome (strain MG1655, EBI Accession U00096.3) and cell volume of 1.03×10-12 mL (72).</caption>
<colgroup>
<col style="width: 8%">
<col style="width: 52%">
<col style="width: 20%">
<col style="width: 5%">
<col style="width: 12%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Variable</th>
<th style="text-align: left;">Definition</th>
<th style="text-align: left;">Starting Value</th>
<th style="text-align: left;">Units</th>
<th style="text-align: left;">Region</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">eGlc</td>
<td style="text-align: left;">External glucose</td>
<td style="text-align: left;">3.1x10-4 to 6.2x10-3</td>
<td style="text-align: left;">M</td>
<td style="text-align: left;">1 mL culture</td>
</tr>
<tr class="even">
<td style="text-align: left;">iGlc</td>
<td style="text-align: left;">Internal glucose</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">M</td>
<td style="text-align: left;">Cytoplasm</td>
</tr>
<tr class="odd">
<td style="text-align: left;">dGTP</td>
<td style="text-align: left;">Deoxyguanosine triphosphate</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">M</td>
<td style="text-align: left;">Cytoplasm</td>
</tr>
<tr class="even">
<td style="text-align: left;">DNA</td>
<td style="text-align: left;">Guanine nucleotides in thesynthesised strand</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">M</td>
<td style="text-align: left;">Cytoplasm</td>
</tr>
<tr class="odd">
<td style="text-align: left;">wtCell</td>
<td style="text-align: left;">Guanine nucleotides in thestrand</td>
<td style="text-align: left;">8.5x10-12</td>
<td style="text-align: left;">M</td>
<td style="text-align: left;">1 mL culture</td>
</tr>
<tr class="even">
<td style="text-align: left;">ROS</td>
<td style="text-align: left;">Reactive oxygen species (H2O2)</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">M</td>
<td style="text-align: left;">Cytoplasm</td>
</tr>
<tr class="odd">
<td style="text-align: left;">odGTP</td>
<td style="text-align: left;">8-oxo-2’-deoxyguanosine</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">M</td>
<td style="text-align: left;">Cytoplasm</td>
</tr>
<tr class="even">
<td style="text-align: left;">mDNA</td>
<td style="text-align: left;">odGTP nucleotides in thesynthesised DNA strand</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">M</td>
<td style="text-align: left;">Cytoplasm</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mCell</td>
<td style="text-align: left;">odGTP nucleotides in theDNA strand</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">M</td>
<td style="text-align: left;">1 mL culture</td>
</tr>
<tr class="even">
<td style="text-align: left;">cytVol</td>
<td style="text-align: left;">Total cytoplasmic volume</td>
<td style="text-align: left;">2.25x10-9</td>
<td style="text-align: left;">mL</td>
<td style="text-align: left;">NA</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<p>To generate hypotheses for the mechanisms of density-associated mutation rate plasticity we constructed a system of ordinary differential equations (ODEs) that recapitulates the dynamics of metabolism, growth and mutagenesis in a 1mL batch culture of <em>E. coli</em> <a href="#fig-one">Figure&nbsp;<span class="quarto-unresolved-ref">fig-one</span></a>. The enzyme MutT, responsible for degrading mutagenic oxidised GTP <span class="citation" data-cites="Maki1992">(<a href="#ref-Maki1992" role="doc-biblioref">Maki and Sekiguchi 1992</a>)</span>, is essential in DAMP <span class="citation" data-cites="Krasovec2017">(<a href="#ref-Krasovec2017" role="doc-biblioref">Krašovec et al. 2017</a>)</span>; the ODE model is therefore focussed on guanine bases. In the model external glucose (<strong><em>eGlc</em></strong>) is taken up by a small initial <em>E. coli</em> population (<strong><em>wtCell</em></strong>). Internal glucose (<strong><em>iGlc</em></strong>) is then metabolised to produce <strong><em>ROS</em>, <em>dGTP</em></strong> and, largely, ‘other’ molecules (‘Sink’ in <a href="#fig-one">Figure&nbsp;<span class="quarto-unresolved-ref">fig-one</span></a>). <strong><em>dGTP</em></strong> is then either integrated into a newly synthesised DNA molecule (<strong><em>DNA</em></strong>) or it reacts with <strong><em>ROS</em></strong> to produce 8-oxo-2’-deoxyguanosine triphosphate (<strong><em>odGTP</em></strong>). In this model, non-oxidised <strong><em>dGTP</em></strong> always pairs correctly with cytosine, producing non-mutant DNA (<strong><em>DNA</em></strong>). In a second round of DNA replication the guanine base is now on the template strand, cytosine is correctly inserted opposite producing new chromosomes (<strong><em>wtCell</em></strong>). <strong><em>odGTP</em></strong>, if it is not dephosphorylated by MutT into <em>dGMP</em> (Sink), can either pair correctly with cytosine (becoming <strong><em>DNA</em></strong>) or mis-pair with adenine (becoming <strong><em>mDNA</em></strong>). When <strong><em>odGTP</em></strong> is inserted opposite adenine into DNA (<strong><em>mDNA</em></strong>) it may be repaired by the MutS or MutY proteins, converting the <strong><em>mDNA</em></strong> back to <strong><em>DNA</em></strong>. The key output of interest is the mutation rate, which is defined as the number of mutant base pairs (<strong><em>mCell</em></strong>) divided by the number of non-mutant base pairs (<strong><em>wtCell</em></strong>). The model comprises 10 ordinary differential equations (ODEs), one for each substance variable in <a href="#fig-one">Figure&nbsp;<span class="quarto-unresolved-ref">fig-one</span></a> (excluding ‘Sink’), plus <strong><em>cytVol</em></strong>, the total population cytoplasmic volume within which all the reactions occur (<a href="#tbl-one">Table&nbsp;<span class="quarto-unresolved-ref">tbl-one</span></a> ,Eq. 1-<a href="https://www.biorxiv.org/content/10.1101/2023.09.27.557722v1.full#disp-formula-10"><strong>10</strong></a>, Methods). These equations require 14 parameters (some of them composite, <a href="https://www.biorxiv.org/content/10.1101/2023.09.27.557722v1.full#T2"><strong>Table 2</strong></a>); the structure and parameter values are largely taken from the existing literature (for details see Methods). Un-measurable parameters (notably the rate of <strong><em>dGTP</em></strong> oxidation to <strong><em>odGTP</em></strong> by <strong><em>ROS</em></strong>, ’<strong>O2’</strong>) were set to give the observed mutation rate (2 × 10<sup>-10</sup> mutations per base pair per generation, (<a href="https://www.biorxiv.org/content/10.1101/2023.09.27.557722v1.full#ref-20"><strong>20</strong></a>)) at a final population density of 3 × 10<sup>8</sup> CFU ml<sup>-1</sup>, typical of 250 mg L<sup>-1</sup> glucose in minimal media. As with most experiments demonstrating density-associated mutation rate plasticity (<a href="https://www.biorxiv.org/content/10.1101/2023.09.27.557722v1.full#ref-3"><strong>3</strong></a>, <a href="https://www.biorxiv.org/content/10.1101/2023.09.27.557722v1.full#ref-21"><strong>21</strong></a>), final population density is controlled by varying initial external glucose. We initiated 28h simulations of 1ml cultures with 2175 cells (a small number, typical of fluctuation assays estimating mutation rate, <a href="https://www.biorxiv.org/content/10.1101/2023.09.27.557722v1.full#F14"><strong>Fig. S10</strong></a>), no internal metabolites and external glucose concentrations relevant to wet-lab experiments – across a log scale from 55 to 1100 mg L<sup>-1</sup> (<a href="https://www.biorxiv.org/content/10.1101/2023.09.27.557722v1.full#T1"><strong>Table 1</strong></a>). The dynamics of external glucose, population size and mutation rate for these simulations are shown in <a href="https://www.biorxiv.org/content/10.1101/2023.09.27.557722v1.full#F1"><strong>Fig.1B-D</strong></a>.</p>


<div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-Alam2016" class="csl-entry" role="listitem">
Alam, Md&nbsp;Kausar, Areej Alhhazmi, John&nbsp;F. DeCoteau, Yu Luo, and C.&nbsp;Ronald Geyer. 2016. <span>“RecA Inhibitors Potentiate Antibiotic Activity and Block Evolution of Antibiotic Resistance.”</span> <em>Cell Chemical Biology</em> 23 (3): 381–91. <a href="https://doi.org/10.1016/j.chembiol.2016.02.010">https://doi.org/10.1016/j.chembiol.2016.02.010</a>.
</div>
<div id="ref-Chu2018" class="csl-entry" role="listitem">
Chu, Xiao-Lin, Bo-Wen Zhang, Quan-Guo Zhang, Bi-Ru Zhu, Kui Lin, and Da-Yong Zhang. 2018. <span>“Temperature Responses of Mutation Rate and Mutational Spectrum in an Escherichia Coli Strain and the Correlation with Metabolic Rate.”</span> <em>BMC Evolutionary Biology</em> 18 (1). <a href="https://doi.org/10.1186/s12862-018-1252-8">https://doi.org/10.1186/s12862-018-1252-8</a>.
</div>
<div id="ref-Cirz2005" class="csl-entry" role="listitem">
Cirz, Ryan T, Jodie K Chin, David R Andes, Valérie de Crécy-Lagard, William A Craig, and Floyd E Romesberg. 2005. <span>“Inhibition of Mutation and Combating the Evolution of Antibiotic Resistance.”</span> Edited by Matt Waldor. <em>PLoS Biology</em> 3 (6): e176. <a href="https://doi.org/10.1371/journal.pbio.0030176">https://doi.org/10.1371/journal.pbio.0030176</a>.
</div>
<div id="ref-Cornick2004" class="csl-entry" role="listitem">
Cornick, N. A., and A. F. Helgerson. 2004. <span>“Transmission and Infectious Dose of <span><em>Escherichia Coli</em></span> O157:H7 in Swine.”</span> <em>Applied and Environmental Microbiology</em> 70 (9): 5331–35. <a href="https://doi.org/10.1128/aem.70.9.5331-5335.2004">https://doi.org/10.1128/aem.70.9.5331-5335.2004</a>.
</div>
<div id="ref-Domenech2020" class="csl-entry" role="listitem">
Domenech, Arnau, Ana Rita Brochado, Vicky Sender, Karina Hentrich, Birgitta Henriques-Normark, Athanasios Typas, and Jan-Willem Veening. 2020. <span>“Proton Motive Force Disruptors Block Bacterial Competence and Horizontal Gene Transfer.”</span> <em>Cell Host &amp; Microbe</em> 27 (4): 544–555.e3. <a href="https://doi.org/10.1016/j.chom.2020.02.002">https://doi.org/10.1016/j.chom.2020.02.002</a>.
</div>
<div id="ref-Foster2007" class="csl-entry" role="listitem">
Foster, Patricia L. 2007. <span>“Stress-Induced Mutagenesis in Bacteria.”</span> <em>Critical Reviews in Biochemistry and Molecular Biology</em> 42 (5): 373–97. <a href="https://doi.org/10.1080/10409230701648494">https://doi.org/10.1080/10409230701648494</a>.
</div>
<div id="ref-Gifford2023" class="csl-entry" role="listitem">
Gifford, Danna R., Anish Bhattacharyya, Alexandra Geim, Rok Marshall Eleanor andKrašovec, and Christopher G. Knight. 2023. <span>“Environmental and Genetic Influence on Rate and Spectrum of Spontaneous Mutations in Escherichia Coli.”</span> <em>biorXiv</em>, June. <a href="https://doi.org/10.1101/2023.04.06.535897">https://doi.org/10.1101/2023.04.06.535897</a>.
</div>
<div id="ref-Imlay2003" class="csl-entry" role="listitem">
Imlay, James A. 2003. <span>“Pathways of Oxidative Damage.”</span> <em>Annual Review of Microbiology</em> 57 (1): 395–418. <a href="https://doi.org/10.1146/annurev.micro.57.030502.090938">https://doi.org/10.1146/annurev.micro.57.030502.090938</a>.
</div>
<div id="ref-Krasovec2018" class="csl-entry" role="listitem">
Krašovec, Rok, Huw Richards, Danna R. Gifford, Roman V. Belavkin, Alastair Channon, Elizabeth Aston, Andrew J. McBain, and Christopher G. Knight. 2018. <span>“Opposing Effects of Final Population Density and Stress on Escherichia Coli Mutation Rate.”</span> <em>The ISME Journal</em> 12 (12): 2981–87. <a href="https://doi.org/10.1038/s41396-018-0237-3">https://doi.org/10.1038/s41396-018-0237-3</a>.
</div>
<div id="ref-Krasovec2017" class="csl-entry" role="listitem">
Krašovec, Rok, Huw Richards, Danna R. Gifford, Charlie Hatcher, Katy J. Faulkner, Roman V. Belavkin, Alastair Channon, Elizabeth Aston, Andrew J. McBain, and Christopher G. Knight. 2017. <span>“Spontaneous Mutation Rate Is a Plastic Trait Associated with Population Density Across Domains of Life.”</span> Edited by Jeff Gore. <em>PLOS Biology</em> 15 (8): e2002731. <a href="https://doi.org/10.1371/journal.pbio.2002731">https://doi.org/10.1371/journal.pbio.2002731</a>.
</div>
<div id="ref-Liu2019" class="csl-entry" role="listitem">
Liu, Haoxuan, and Jianzhi Zhang. 2019. <span>“Yeast Spontaneous Mutation Rate and Spectrum Vary with Environment.”</span> <em>Current Biology</em> 29 (10): 1584–1591.e3. <a href="https://doi.org/10.1016/j.cub.2019.03.054">https://doi.org/10.1016/j.cub.2019.03.054</a>.
</div>
<div id="ref-Loewe2003" class="csl-entry" role="listitem">
Loewe, Laurence, Volker Textor, and Siegfried Scherer. 2003. <span>“High Deleterious Genomic Mutation Rate in Stationary Phase of <span><em>Escherichia Coli</em></span>.”</span> <em>Science</em> 302 (5650): 1558–60. <a href="https://doi.org/10.1126/science.1087911">https://doi.org/10.1126/science.1087911</a>.
</div>
<div id="ref-MacLean2013" class="csl-entry" role="listitem">
MacLean, R. Craig, Clara Torres-Barceló, and Richard Moxon. 2013. <span>“Evaluating Evolutionary Models of Stress-Induced Mutagenesis in Bacteria.”</span> <em>Nature Reviews Genetics</em> 14 (3): 221–27. <a href="https://doi.org/10.1038/nrg3415">https://doi.org/10.1038/nrg3415</a>.
</div>
<div id="ref-Maharjan2017" class="csl-entry" role="listitem">
Maharjan, Ram P., and Thomas Ferenci. 2017. <span>“A Shifting Mutational Landscape in 6 Nutritional States: Stress-Induced Mutagenesis as a Series of Distinct Stress Input<span></span>mutation Output Relationships.”</span> Edited by Jeff Gore. <em>PLOS Biology</em> 15 (6): e2001477. <a href="https://doi.org/10.1371/journal.pbio.2001477">https://doi.org/10.1371/journal.pbio.2001477</a>.
</div>
<div id="ref-Maharjan2018" class="csl-entry" role="listitem">
Maharjan, Ram P, and Thomas Ferenci. 2018. <span>“The Impact of Growth Rate and Environmental Factors on Mutation Rates and Spectra in Escherichia Coli.”</span> <em>Environmental Microbiology Reports</em> 10 (6): 626–33.
</div>
<div id="ref-Maki1992" class="csl-entry" role="listitem">
Maki, Hisaji, and Mutsuo Sekiguchi. 1992. <span>“MutT Protein Specifically Hydrolyses a Potent Mutagenic Substrate for DNA Synthesis.”</span> <em>Nature</em> 355 (6357): 273–75. <a href="https://doi.org/10.1038/355273a0">https://doi.org/10.1038/355273a0</a>.
</div>
<div id="ref-Ragheb2019" class="csl-entry" role="listitem">
Ragheb, Mark N., Maureen K. Thomason, Chris Hsu, Patrick Nugent, John Gage, Ariana N. Samadpour, Ankunda Kariisa, et al. 2019. <span>“Inhibiting the Evolution of Antibiotic Resistance.”</span> <em>Molecular Cell</em> 73 (1): 157–165.e5. <a href="https://doi.org/10.1016/j.molcel.2018.10.015">https://doi.org/10.1016/j.molcel.2018.10.015</a>.
</div>
<div id="ref-Tenaillon2010" class="csl-entry" role="listitem">
Tenaillon, Olivier, David Skurnik, Bertrand Picard, and Erick Denamur. 2010. <span>“The Population Genetics of Commensal Escherichia Coli.”</span> <em>Nature Reviews Microbiology</em> 8 (3): 207–17. <a href="https://doi.org/10.1038/nrmicro2298">https://doi.org/10.1038/nrmicro2298</a>.
</div>
</div>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./Introduction.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./Discussion.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Discussion</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>